test/modifiers/batch.test.js

```diff
@@ -0,0 +1,284 @@
+        });
+    });
+  });
+
+  describe('when both "time" and "count" are defined in options', function () {
+    function asyncWrite(stream, dataArr) {
+      if (dataArr.length) {
+        setTimeout(function () {
+          stream.push(dataArr[0]);
+
+          asyncWrite(stream, dataArr.slice(1));
+        }, 1);
+      } else {
+        stream.end();
+      }
+    }
+
+    it('writes chunks if count is met first before the defined time', function (done) {
+      var input = through.obj();
+      var chunks = 0;
+      var CHUNKS = 10;
+
+      input
+        .pipe(batch({
+          count: 1,
+          time: 5000
+        }))
+        .on('data', function (chunk) {
+          chunks += 1;
+          expect(chunk).to.be.an('array').and.to.have.lengthOf(1);
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunks).to.equal(CHUNKS);
+          done();
+        });
+
+      asyncWrite(input, _.times(CHUNKS));
+    });
+
+    it('writes chunks if time is met first before the defined count', function (done) {
+      var input = through.obj();
+      var CHUNKS = 10;
+      var reads = 0;
+
+      input
+        .pipe(batch({
+          count: 100,
+          time: 1
+        }))
+        .on('data', function (chunk) {
+          reads += 1;
+          expect(chunk).to.be.an('array').and.to.have.length.above(0);
+        })
+        .on('error', done)
+        .on('end', function () {
+          // for good measure, leave some wiggle room as to
+          // how many writes we expect, because timers
+          expect(reads).to.be.above(CHUNKS / 2);
+          done();
+        });
+
+      asyncWrite(input, _.times(10));
+    });
+
+    it('successfully ends if there is no data written', function (done) {
+      var input = getReadableStream([], {
+        objectMode: true
+      });
+
+      var chunksRead = false;
+
+      input
+        .pipe(batch({
+          count: 2,
+          time: 1
+        }))
+        .on('data', function (chunk) {
+          chunksRead = true;
+          expect(chunk).to.be.an('array');
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunksRead).to.equal(false);
+          done();
+        });
+    });
+  });
+
+  describe('when neither "time" nor "count" are defined in options', function () {
+    var expected = data.map(function (val) {
+      return [buff(val)];
+    });
+    var objExpected = objData.map(function (val) {
+      return [val];
+    });
+
+    basicTests(null, expected, objExpected);
+
+    it('writes chunks immediately when read', function (done) {
+      var DATA = _.times(5);
+      var input = through.obj();
+      var idx = 0;
+      var chunksRead = false;
+
+      function write() {
+        input.push(DATA[idx]);
+        idx += 1;
+      }
+
+      input
+        .pipe(batch())
+        .on('data', function (chunk) {
+          chunksRead = true;
+          expect(chunk).to.be.an('array');
```

Comment:
Correct. Originally, I thought it would pass through all the elements as-is (i.e. not in an array), but really, that would actually be different behavior. In reality, no options means it should run with default options, and the default options would be something similar to `{ count: 1 }` (count of 0 makes no sense, I think) or `{ time: 0 }`, both of which basically result in writing data immediately. However, it would still apply a batch to it. This way, `batch` will always transform the stream from items to array of items, so you can safely always assume that you can handle a `batch` stream as arrays, and you can always flatten it afterwards, without the risk of messing up if the stream happens to be an object stream of arrays to begin with. So... consistency.

Do you think better docs are needed for this?